### PR TITLE
New gadget, bug fixing, typo fixing

### PR DESCRIPTION
**Pull request by irsdl**
SessionSecurityToken gadget has been added. I got this from my old note (I hadn't fully explored it at the time) but I am not sure whether this is the same as what @mufinnnnnnn in Twitter had published and removed (the blog post was removed shortly after so it was not possible to have a look).

SerializersHelper has been enhanced (useful when developing a new gadget or plugin).

Some bugs have been fixed in the minification area.

Some typos have been addressed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,50 +24,53 @@ This software is a personal project and not related with any companies, includin
...
 
 Available gadgets:
 
 	ActivitySurrogateDisableTypeCheck (Disables 4.8+ type protections for ActivitySurrogateSelector, command is ignored.)
 		Formatters:
-			BinaryFormatter, ObjectStateFormatter, SoapFormatter, NetDataContractSerializer, LosFormatter
+			BinaryFormatter, LosFormatter, NetDataContractSerializer, ObjectStateFormatter, SoapFormatter
 
-	ActivitySurrogateSelectorFromFile (Another variant of the ActivitySurrogateSelector gadget. This gadget interprets the command parameter as path to the .cs file that should be compiled as exploit class. Use semicolon to separate the file from additionally required assemblies, e. g., '-c ExploitClass.cs;System.Windows.Forms.dll'.)
+	ActivitySurrogateSelector (This gadget ignores the command parameter and executes the constructor of ExploitClass class.)
```
Comment (pwntester) [inline README.md]: Why removing this info?

File: README.md [hunk 2]
```diff
@@ -24,50 +24,53 @@ This software is a personal project and not related with any companies, includin
...
 		Formatters:
-			BinaryFormatter, ObjectStateFormatter, SoapFormatter, LosFormatter
+			BinaryFormatter, LosFormatter, ObjectStateFormatter, SoapFormatter
 
-	ActivitySurrogateSelector (This gadget ignores the command parameter and executes the constructor of ExploitClass class.)
+	ActivitySurrogateSelectorFromFile (Another variant of the ActivitySurrogateSelector gadget. This gadget interprets the command parameter as path to the .cs file that should be compiled as exploit class. Use semicolon to separate the file from additionally required assemblies, e. g., '-c ExploitClass.cs;System.Windows.Forms.dll'.)
```
Comment (pwntester) [inline README.md]: same here

File: README.md [hunk 3]
```diff
@@ -24,50 +24,53 @@ This software is a personal project and not related with any companies, includin
...
+			DataContractSerializer, FastJson, FsPickler, JavaScriptSerializer, Json.Net, Xaml, XmlSerializer, YamlDotNet < 5.0.0
 
-	TextFormattingRunProperties (TextFormattingRunProperties gadget)
+	PSObject (PSObject gadget. Target must run a system not patched for CVE-2017-8565 (Published: 07/11/2017))
 		Formatters:
-			BinaryFormatter, ObjectStateFormatter, SoapFormatter, NetDataContractSerializer, LosFormatter
+			BinaryFormatter, LosFormatter, NetDataContractSerializer, ObjectStateFormatter, SoapFormatter
 
-	PSObject (PSObject gadget. Target must run a system not patched for CVE-2017-8565 (Published: 07/11/2017))
+	SessionSecurityToken (SessionSecurityTokenGenerator (System.IdentityModel.Tokens namespace) gadget)
```
Comment (pwntester) [inline README.md]: do we need the namespace info here?

File: README.md [hunk 4]
```diff
@@ -195,21 +198,25 @@ Special thanks to all contributors:
...
+	ActivitySurrogateSelectorFromFile
+		James Forshaw
 	ObjectDataProvider
 		Oleksandr Mirosh and Alvaro Munoz
-	TextFormattingRunProperties
-		Oleksandr Mirosh and Alvaro Munoz
 	PSObject
 		Oleksandr Mirosh and Alvaro Munoz
+	SessionSecurityToken
+		Soroush Dalili
```
Comment (pwntester) [inline README.md]: maybe mention @mfnnnnnnn ? that was the gadget he mentioned in his blog post

File: README.md (hunk 3 above)
Comment (irsdl) [reply README.md]: I just didn't want to make it too confusing - I also think we may want to add this info as an additional bit to all the gadgets so people would know which namespace they are dealing with? We can always remove it in the future releases if we find it useless.

**Comment by irsdl** [comment]
all done,  ready for approval

**Pull request merged**